File: 10. Wild pointer demo.cpp
// Topic: Wild pointer demonstration
// Description: Show uninitialized pointer (wild) vs properly initialized to NULL
#include <iostream>
using namespace std;

int main()
{
    int* wild;  // uninitialized - wild pointer (contains garbage)
    // cout << *wild; // DO NOT dereference - undefined behavior
    int* safe = NULL;
    if (safe)
        cout << *safe << '\n';
    else
        cout << "safe is NULL, not dereferencing" << '\n';
    return 0;
}

// Time Complexity: O(1).
// Space Complexity: O(1).

File: 10. Wild pointer demo.py
"""
Topic: Wild pointer demonstration (Python explanation)
Description: Python variables are always initialized; uninitialized pointer concept doesn't apply.
"""
def main():
    try:
        print(undefined_variable)  # NameError
    except NameError:
        print('No wild pointers in Python; uninitialized names raise NameError')

if __name__ == '__main__':
    main()
    # Time Complexity: O(1).
    # Space Complexity: O(1).
